internal/controllers/routerinterface/reconcile.go

```diff
@@ -143,64 +128,59 @@ func (r *orcRouterInterfaceReconciler) getNetworkClient(ctx context.Context, obj
 	return clientScope.NewNetworkClient()
 }
 
-func (r *orcRouterInterfaceReconciler) reconcileNormal(ctx context.Context, router *orcv1alpha1.Router, routerInterface *orcv1alpha1.RouterInterface, routerInterfacePorts []ports.Port, networkClient osclients.NetworkClient) (_ time.Duration, err error) {
+func (r *orcRouterInterfaceReconciler) reconcileNormal(ctx context.Context, router *orcv1alpha1.Router, routerInterface *orcv1alpha1.RouterInterface, routerInterfacePorts []ports.Port, networkClient osclients.NetworkClient) (reconcileStatus progress.ReconcileStatus) {
 	log := ctrl.LoggerFrom(ctx)
 	log.V(logging.Verbose).Info("Reconciling router interface", "name", routerInterface.Name)
 
-	var statusOpts updateStatusOpts
+	var osResource *ports.Port
 
 	// Ensure we always update status
 	defer func() {
-		if err != nil {
-			statusOpts.err = err
-		}
+		reconcileStatus = reconcileStatus.WithReconcileStatus(
+			r.updateStatus(ctx, routerInterface, osResource, reconcileStatus))
 
-		err = errors.Join(err, r.updateStatus(ctx, routerInterface, &statusOpts))
+		if needsReschedule, _ := reconcileStatus.NeedsReschedule(); !needsReschedule && osResource != nil {
+			log.V(logging.Status).Info("Router interface is available")
+		}
 
+		// Don't return a terminal error because we don't aggregate them
+		err := reconcileStatus.Error()
 		var terminalError *orcerrors.TerminalError
 		if errors.As(err, &terminalError) {
-			log.V(logging.Info).Info("not scheduling further reconciles for terminal error", "err", err.Error())
-			err = nil
+			reconcileStatus = nil
 		}
 	}()
 
-	var poll bool
-	var createOpts routers.AddInterfaceOptsBuilder
 	switch routerInterface.Spec.Type {
 	case orcv1alpha1.RouterInterfaceTypeSubnet:
-		poll, createOpts, err = r.reconcileNormalSubnet(ctx, routerInterface, routerInterfacePorts, &statusOpts)
+		osResource, reconcileStatus = r.reconcileNormalSubnet(ctx, router, routerInterface, routerInterfacePorts, networkClient)
+
 	default:
-		err = orcerrors.Terminal(orcv1alpha1.ConditionReasonInvalidConfiguration, fmt.Sprintf("Invalid type %s", routerInterface.Spec.Type))
+		return progress.WrapError(
+			orcerrors.Terminal(orcv1alpha1.ConditionReasonInvalidConfiguration, fmt.Sprintf("Invalid type %s", routerInterface.Spec.Type)))
```

Comment:
Won't this error be dropped by the defer function?

Reply:
Yes, but it will be handled in updateStatus before it's dropped.